x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_issues_step/components/kibana_deprecation_issues_panel.tsx

```diff
@@ -30,20 +30,26 @@ export const KibanaDeprecationIssuesPanel: FunctionComponent<Props> = ({ setIsFi
   const [error, setError] = useState<Error | undefined>(undefined);
 
   useEffect(() => {
+    let isMounted = true;
```

Comment:
@gsoldevila It looks like these changes address a race condition in which the component is unmounted before `getAllDeprecations` resolves. Is that right? Can you help me understand the background to these changes? What was the behavior you originally observed and how did you trace it back to this file? Thanks!

Reply:
Good question, I don't remember exactly, but at some point one of the tests was failing and I was having an error on the lines of [Can't perform a React state update on an unmounted component](https://stackoverflow.com/questions/53949393/cant-perform-a-react-state-update-on-an-unmounted-component).

I believe the test was finishing (and unmounting the component) before the call to `await deprecations.getAllDeprecations()`.